### PR TITLE
Add log sources command

### PR DESCRIPTION
**Pull request by DuncanUszkay1**
Fixes https://github.com/Shopify/shopify-functions/issues/373

Adds the `shopify app logs sources` command, a listing of the sources that are valid arguments for `shopify app logs --source=[csv list of valid sources]`

Tophat 🎩 

Help Menu:
```
macbook-pro:cli duncanuszkay$ pnpm run shopify help app logs sources
Print out a list of sources to use with the logs command.

USAGE
  $ shopify app logs sources [-c <value>] [--no-color] [--path <value>] [--verbose]

FLAGS
  -c, --config=<value>  The name of the app configuration.
      --no-color        Disable color output.
      --path=<value>    The path to your app directory.
      --verbose         Increase the verbosity of the output.

DESCRIPTION
  Print out a list of sources to use with the logs command.

  Currently only functions are supported as sources.
```

Command output:
```
macbook-pro:cli duncanuszkay$ pnpm run shopify app logs sources --path=cv
╭─ info ───────────────────────────────────────────────────────────────────────╮
│                                                                              │
│  You are running a global installation of Shopify CLI                        │
│                                                                              │
│  This project has Shopify CLI as a local dependency in package.json. If you  │
│   prefer to use that version, run the command with your package manager      │
│  (e.g. pnpm run shopify).                                                    │
│                                                                              │
│  For more information, see Shopify CLI documentation                         │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

EXTENSIONS                         
extensions.10-percent-off
extensions.cv
```

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/app/src/cli/commands/app/app-logs/sources.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import {appFlags} from '../../../flags.js'
+import {AppInterface} from '../../../models/app/app.js'
+import {loadApp} from '../../../models/app/loader.js'
+import Command from '../../../utilities/app-command.js'
+import {loadLocalExtensionsSpecifications} from '../../../models/extensions/load-specifications.js'
+import {sources} from '../../../services/app-logs/sources.js'
+import {globalFlags} from '@shopify/cli-kit/node/cli'
+
+export default class Sources extends Command {
+  static summary = 'Print out a list of sources to use with the logs command.'
```
Comment (nickwesselman) [inline packages/app/src/cli/commands/app/app-logs/sources.ts:10]:
```suggestion
  static summary = 'Print out a list of sources that may be used with the logs command.'
```

File: packages/app/src/cli/commands/app/app-logs/sources.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  static descriptionWithMarkdown = `Currently only function extensions are supported as sources.`
```
Comment (nickwesselman) [inline packages/app/src/cli/commands/app/app-logs/sources.ts:12]:
```suggestion
  static descriptionWithMarkdown = `The output source names can be used with the \`--source\` argument of \`shopify app logs\` to filter log output. Currently only function extensions are supported as sources.`
```

**Review by nickwesselman — APPROVED** [review]

**Comment by nickwesselman** [comment]
A couple suggestions on help/docs for the command but otherwise lgtm!

**Review by lopert — APPROVED** [review]
🎩 worked for me!

**Pull request merged**